TasksApp/TasksApp/PriorityTask.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+namespace TasksApp
+{
+    internal class PriorityTask : WeeklyTask
```

Comment:
wrong inheritance - you have `Date` info duplicated in 2 classes, this is wrong
```suggestion
    internal class PriorityTask : RegularTask
```